Hw4_Module/Hw4_Module/Program.cs

```diff
@@ -0,0 +1,258 @@
+            }
+            else 
+            {
+                SearchFreeSpaceInArray(ref itemsArray, item);
+                SearchFreeSpaceInArray(ref itemsArray, item, "Add");
+                Console.WriteLine($"Item - {item} add");
+            }
+        }
+
+        public static void RemoveItem(ref string[,] itemsArray)
+        {
+            Console.WriteLine("Enter your item or *");
+            string item = Console.ReadLine();
+            if (item == "*")
+            {
+                Array.Clear(itemsArray, 0, itemsArray.Length);
+                Console.WriteLine("Array clear");
+            }
+            else 
+            {
+                bool check = CheckItemInArrayAddOrRemove(ref itemsArray, item, "Remove");
+                if (check == true)
+                {
+                    Console.WriteLine($"Item - {item} removed");
+                }
+                else 
+                {
+                    Console.WriteLine($"Item - {item} don't exist");
+                }
+            }
+        }
+
+        public static void MarkAs(ref string[,] itemsArray)
+        {
+            Console.WriteLine("Enter your item");
+            string item = Console.ReadLine();
+            Console.WriteLine("Enter your status");
+            string status = Console.ReadLine();
+            if (status != "0" && status != "1")
+            {
+                Console.WriteLine("incorect status");
+                return;
+            }
+            else if (status == "1") 
+            {
+                Console.WriteLine("Enter your date or press Enter");
+                string date = Console.ReadLine();
+                if (date == "") 
+                {
+                    DateTime dateT = DateTime.Now;
+                    date = dateT.ToString("dddd, dd MMMM yyyy");
+                }
+                status = status + " " + date;
+                Console.WriteLine($"status {status}");
+            }
+
+            bool checkItem = CheckItemInArrayAddOrRemove(ref itemsArray, item, "Search", status);
+            if (checkItem == true)
+            {
+                CheckItemInArrayAddOrRemove(ref itemsArray, item, "ChangeStatus", status);
+                Console.WriteLine($"{item} status changed to {status}");
+                return;
+            }
+            else 
+            {
+                Console.WriteLine($"Item - {item} don't exist");
+            }
+        }
+
+        public static void ShowItems(ref string[,] itemsArray)
+        {
+            Console.WriteLine("Enter your status");
+            string status = Console.ReadLine();
+            if (status != "0" && status != "1" && status!="")
+            {
+                Console.WriteLine("incorect status");
+                return;
+            }
+            ShowItemsFromArray(ref itemsArray, status);
+        }
+
+        public static bool CheckItemInArrayAddOrRemove(ref string[,] itemsArray, string item,
```

Comment:
function should have one reason to change. Here You combined search, addition, and removal. Search should be separate, in case you need an index of found item just return it.